feat: added pathpoint documentation (#12027)
newrelic/docs-website: src/content/docs/business-observability/troubleshoot-pathpoint.mdx

  - Pathpoint
  - Troubleshoot Pathpoint
metaDescription: How to troubelshoot a pathpoint using special filters i.e. Canary, Flame filter.
---

import stagesTypes from 'images/business-observability-screenshot-stages-types.png'

import canaryFilter from 'images/business-observability-screenshot-canary-filter.png'

import activeCanaryFilter from 'images/business-observability-screenshot-active-canary-filter.png'

import pathpointConfig from 'images/business-observability-screenshot-pathpoint-config.png'

import generalConfig from 'images/business-observability-screenshot-pathpoint-general-config.png'

import credentials from 'images/business-observability-screenshot-pathpoint-credentials.png'

import installJob from 'images/business-observability-screenshot-install-job.png'

import activateFlameFilter from 'images/business-observability-screenshot-activate-flame-filter.png'

import activatedFlameFilter from 'images/business-observability-screenshot-activated-flame-filter.png'

import activateDropFilter from 'images/business-observability-screenshot-activate-drop-filter.png'

import activatedDropFilter from 'images/business-observability-screenshot-activated-drop-filter.png'

> Pathpoint allows you to execute different troubleshooting tasks using New Relic Synthetic monitors. These tasks can be run in the background allowing you to proactively identify and resolve issues before they have a significant impact on your customer experience. 

Reviewer: ```suggestion
Pathpoint allows you to execute different troubleshooting tasks using New Relic Synthetic monitors. You can run these tasks in the background so you can proactively identify and resolve issues before they have a significant impact on your customer experience. 
```

On this doc, you learn to trouble your Pathpoint using:

- Canary filter
- Flame filter
- Drop filter

> ## Canary filter

Reviewer: I'd divide the content here into steps.

Author: Do you mean the steps of applying the filter?

Author: already did! :)

> The Canary filter is a highly effective tool for troubleshooting your Pathpoint implementation. When activated, it limits Pathpoint's calculations to only the steps that are manually activated, allowing you to pinpoint and understand the specific operation of a particular step. 

Reviewer: I'd add a screenshot of the logo.

This feature can be especially helpful when troubleshooting issues with a specific step, during code deployment or system maintenance, when you need to narrow your view. 

<img
    title="Types of stages"
    alt="types of stages"
    src={stagesTypes}
/>

On this Pathpoint, you can see all the different steps that are active right now. To troubleshoot this Pathpoint, you can turn off all active steps and then enable them one by one, while also turning on the Canary filter. This approach helps to isolate any problematic steps or touchpoints in the customer journey, allowing for targeted debugging and optimization.

To activate the filter, just click on the Canary Symbol located on the top right part of the Pathpoint window.

<img
    title="Canary Filter"
    alt="Canary Filter"
    src={canaryFilter}
/>

> On the Pop Up window that opens up, click "Continue" to activate this filter. Once active, it changes its color to yellowish. Now, you can click on any of the steps you want to troubleshoot. 

Reviewer: ```suggestion
In the dialog box, click **Continue** to activate this filter. Once it's active, it changes its color to yellow. Now, you can click on any of the steps you want to troubleshoot. 
```

<img
    title="Active Canary Filter"
    alt="Active Canary Filter"
    src={activeCanaryFilter}
/>

> Notice that enabling a step also enables the corresponding touchpoints tied to the step.

Reviewer: ```suggestion
<Callout variant="tip">
Enabling a step also enables the corresponding touchpoints tied to the step.
</Callout>
```

## Flame filter

> The Flame Filter highlights the most problematic “steps and touchpoints” with a configurable time window and percentile threshold. 

Reviewer: ```suggestion
The Flame Filter highlights the most problematic steps and touchpoints with a configurable time window and percentile threshold. 
```

When using this filter, you see the worst-performing touchpoints highlighted in red, making it easy to identify the areas that require attention. For example, the filter can help you identify touchpoints that are not meeting the threshold percentage set in the tuning.
This is very useful in situations where things look okay now, but might have had a pattern of errors or latency in the recent past. 

Before you use this special Pathpoint filter, you must enable **Flame Filter Background Script**.

### Enable Flame Filter Background Script

To enable the Flame Filter Background Script:

<Steps>

<Step>

> Navigate to the Pathpoint screen and click on the three lines in the top left part of the window.

Reviewer: I'd add the breadcrumb to get to the screen.

<img
  title="Navigate to Pathpoint config"
  alt="Navigate to Pathpoint config"
  src={pathpointConfig}
/>

</Step>

<Step>

> Here, select **Credentials and General Configuration**.

Reviewer: ```suggestion
Select **Credentials and General Configuration**.
```

<img
  title="Pathpoint Credentials and General Configuration"
  alt="Pathpoint Credentials and General Configuration"
  src={generalConfig}
/>

</Step>

<Step>

> On the **Credentials and General Configuration** dialog box window, you see a few options asking you to fill **Ingest License** key, as well as a **User API Key**, and options to ** Enable Pathpoint Logging** and **Enable Flame Filter Background Script**.

Reviewer: I'd remove this, since they can see this in the UI.

Once you've filled in the required data and checked both options, click **Save/Update**.

<img
  title="Save/Update Pathpoint Credentials"
  alt="Save/Update Pathpoint Credentials"
  src={credentials}
/>

This takes you back to main Pathpoint window.

</Step>

<Step>

From the main Pathpoint screen, again click on the three lines in the top left part of the window.

> 

Reviewer: I'd add the logo instead of describing it and where its located.

<img
  title="Navigate to Pathpoint config"
  alt="Navigate to Pathpoint config"
  src={pathpointConfig}
/>

</Step>

<Step>

> Now, click **Credentials and General Configuration**.

Reviewer: ```suggestion
Click **Credentials and General Configuration**.
```

<img
  title="Pathpoint Credentials and General Configuration"
  alt="Pathpoint Credentials and General Configuration"
  src={generalConfig}
/>

</Step>

<Step>

Finally, click **Install/Update Job**.

<img
  title="Install/Update Job"
  alt="Install/Update Job"
  src={installJob}
/>

</Step>

Now, both **Pathpoint Logging**, and **Flame Filter Background Script** are enabled.

</Steps>

> Once you've enabled the **Flame Filter Background Script**, you can activate the Flame filter by just clicking on the Flame Symbol.

Reviewer: I'd add the logo instead of describing it. That way, we can also remove the screenshot below.

<img
  title="Activate Flame filter"
  alt="Activate Flame filter"
  src={activateFlameFilter}
/>

On the Pop Up window that opens up, click "Continue" to activate this filter. Once active, the filter changes its color to a reddish flame and highlight the problematic touchpoints in red. 

<img
  title="Activated Flame filter"
  alt="Activated Flame filter"
  src={activatedFlameFilter}
/>

## Drop filter

The Drop filter calculates the order or transaction loss by stage, including the monetary value of these losses based on the values set in the tuning. This filter also enables you to identify the specific steps in the customer journey that are causing these drops.

> To activate it, just click on the tear-drop Symbol.

Reviewer: I'd add a screenshot of the logo instead of describing it here.

> <img

Reviewer: Instead of this screenshot, I'd add a screenshot of the logo next to its name: "The Drop filter LOGO calculates [...]" above.
